### DISCUSSION TITLE
EBGP hosts and route filtering

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#841] [Features]
I'm working on a use case involving FRR EBGP hosts with EVPN, and I'm considering adding the following to filter routes:
```
{% for n in bgp.neighbors if n.evpn|default(False) %}
{%  for af in ['ipv4','ipv6'] if af in n %}
{%   set peer = n[af] if n[af] is string else n.local_if|default('?') %}
  neighbor {{ peer }} activate
{%   if role|default('?') == 'ebgp-host' %}  <--- added
  neighbor {{ peer }} route-map no-export in
{%   endif %}
{%   if bgp.rr|default('') and not n.rr|default('') %}
  neighbor {{ peer }} route-reflector-client
{%   endif %}
{%  endfor %}
{% endfor %}

 exit-address-family
!
exit

{% if role|default('?') == 'ebgp-host' %}
! Stop received routes from being re-advertised
route-map no-export permit 10
 set community no-export
exit
{% endif %}
```

Thoughts or objections? Using just 'role: host' triggers other undesirable side-effects

**Comment by ipspace** [comment]
This is something that can easily be solved with a custom template (probably the easiest approach) or a plugin. I see no reason to add another nerd knob to support a niche use case.

Maybe someone should volunteer to write a generic BGP policy plugin ;)

**Reply by jbemmel** [reply]
Well, technically an additional value for an existing nerd knob - but I understand what you're saying

I can package it as an plug-in, no problem
